src/main/java/pl/touk/sputnik/connector/gerrit/GerritFacadeBuilder.java

```diff
@@ -56,6 +66,27 @@ private GerritPatchset buildGerritPatchset(Configuration configuration) {
         notBlank(changeId, "You must provide non blank Gerrit change Id");
         notBlank(revisionId, "You must provide non blank Gerrit revision Id");
 
-        return new GerritPatchset(changeId, revisionId, tag);
+        return new GerritPatchset(urlEncodeChangeId(changeId), revisionId, tag);
+    }
+
+    public static String urlEncodeChangeId(String changeId) {
+        if (changeId.indexOf('%') >= 0) {
+            // ChangeID is already encoded (otherwise why it would have a '%' character?)
+            return changeId;
+        }
+        // To keep the changeId readable, we don't encode '~' (it is not needed according to RFC2396)
+        return StreamSupport.stream(Splitter.on('~').split(changeId).spliterator(), false)
+                .map(GerritFacadeBuilder::safeUrlEncode).collect(Collectors.joining("~"));
```

Comment:
I just hope you're right and it works :D

Reply:
I will add a comment with a reference to
ChangesRestClient.id where they do it in a very similar way with the 2 and 3 argument methods.